include/onnxruntime/core/framework/execution_provider.h

```diff
@@ -77,7 +77,7 @@ class IExecutionProvider {
   /**
    * Get an allocator with specified device id and MemType. Return nullptr if it doesn't exist
    */
-  virtual AllocatorPtr GetAllocator(int device_id, OrtMemType mem_type) const;
+  virtual AllocatorPtr GetAllocator(OrtMemType mem_type) const;
```

Comment:
The removal of device_id makes this method a bit misleading. It gives me the impression that the output of GetAllocator depends on OrtMemType only. As it stands now it's very clear that an allocator is a function of both device_id and mem_type. What's the disadvantage of keeping the status quo?

Reply:
Alright, given that an EP is associated with a device I suppose this is ok. This does mean that GetDevice() *must* be implemented.